File: Largest Number formed from an Array - GFG/largest-number-formed-from-an-array.java
//{ Driver Code Starts
//Initial Template for Java



import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine().trim());
        while (tc-- > 0) {
            String[] inputLine;
            int n = Integer.parseInt(br.readLine().trim());
            String[] arr = br.readLine().trim().split(" ");

            String ans = new Solution().printLargest(arr);
            System.out.println(ans);
        }
    }
}
// } Driver Code Ends


//User function Template for Java
class numComperator implements Comparator<String>{
     @Override
    public int compare(String a,String b){
        return -(a+b).compareTo(b+a);
    }
}
class Solution {
    String printLargest(String[] arr) {
        List<String>li = new ArrayList<>();
        for(String s : arr)
            li.add(s);
        Collections.sort(li,new numComperator());
        String res = "";
        for(String s:li)
            res += s;
        return res;
    }
}

File: Largest Number formed from an Array - GFG/largest-number-formed-from-an-array.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;


// } Driver Code Ends
//User function template for C++
bool comp(string &s1,string &s2)
{
    return s1+s2>s2+s1;
}
class Solution{
public:
	// The main function that returns the arrangement with the largest value as
	// string.
	// The function accepts a vector of strings
	string printLargest(vector<string> &arr) {
	    // code here
	    sort(arr.begin(),arr.end(),comp);
	    string s;
	    for(auto it:arr)
	    s+=it;
	    return s;
	}
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, i;
        cin >> n;
        vector<string> arr(n);
        for (i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.printLargest(arr);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends